### ISSUE TITLE
Connection troubles with KeePass 2 and KeeAgent

### ISSUE THREAD

**Issue by cblake777** [mhogomchungu/sirikali#150]
I am having issues trying to get SiriKali to connect via KeePass2 and KeeAgent.  I'm using SSHFS-Win to mount the remote endpoint.

This is my configuration:
![image](https://user-images.githubusercontent.com/391509/83476293-52210c00-a45e-11ea-9571-513761767d85.png)

And this is the response from the debug window when trying to connect:
![image](https://user-images.githubusercontent.com/391509/83476390-9e6c4c00-a45e-11ea-8e04-3d101e120760.png)

However, this command put into a batch file connects without issue using KeePass / KeeAgent:
```
C:\Users\Carl Blakemore>set "SSH_AUTH_SOCK=C:\Temp\keeagent.socket" && set "PATH=C:\Program Files\SSHFS-Win\bin" && "C:\Program Files\SSHFS-Win\bin\sshfs.exe" -o idmap=user cblakemore@-------------: W:"

```

Any help would be greatly appreciated.

Thanks!
Carl Blakemore

**Comment by mhogomchungu** [comment]
What version of sshfs-win are you using?

Maybe its an older version and it doesn't know about ```create_file_umask=0000,create_dir_umask=0000,umask=0000``` options.

Remove these options and try again.

Things works here as expected.

**Comment by cblake777** [comment]
I'm running WinFSP 2020, I get exactly the same error message if I eliminate `create_file_umask=0000,create_dir_umask=0000,umask=0000`.

**Comment by mhogomchungu** [comment]
1. What version of sshfs-win are you using?
2. Can you connect using a password or ssh private key?

**Comment by cblake777** [comment]
sshfs-win version: 3.5.20024

```
PS C:\Program Files\SSHFS-Win\bin> .\sshfs.exe --version
SSHFS version 3.5.2
FUSE library version 3.2
```

I can connect using a private key, I also have the mount working using KeePass / KeeAgent via a batch script with the command listed above as well[1], so sshfs is working, it just seems to fail here. I'm certain I'm doing something wrong, I just haven't been able to see it.  

[1] The scripts were borrowed heavily from the initial post here: https://github.com/billziss-gh/sshfs-win/issues/15

**Comment by mhogomchungu** [comment]
I installed ```sshfs-win version: 3.5.20024``` and ```WinFSP 2020``` and i can't reproduce the issue.

In KeyPass, are you setting the socket path is the text field pointed by the arrow?.

![keyagent](https://user-images.githubusercontent.com/3496115/83557848-0718fe80-a51b-11ea-9929-e2cb855257bd.png)

**Comment by cblake777** [comment]
![image](https://user-images.githubusercontent.com/391509/83558829-9bff0600-a4e1-11ea-87a4-3739bff91494.png)

**Comment by mhogomchungu** [comment]
>I can connect using a private key, I also have the mount working using KeePass / KeeAgent via a batch script with the command listed above as well[1], so sshfs is working, it just seems to fail here. I'm certain I'm doing something wrong, I just haven't been able to see it.

The above is a bit ambiguous to me so let me rephrase the question.

1. Can you connect to the remote server using a password with SiriKali?
2. Can you connect to the remote server using a private key with SiriKali?

**Comment by cblake777** [comment]
I've done some more digging into this today.  First, to answer your questions,

**Comment by cblake777** [comment]
I've done some more digging into this today and I think I see a couple problems.

First, to answer your questions:
1. The server I'm trying to connect with doesn't have a password for this account, and I haven't had time to try a different machine just yet.
2.  I cannot connect with a private key

That being said, I think I've identified an issue (or two).

1. In messing around on the command line I kept getting an error "Connection reset by peer".  Further digging into it, I discovered that the "Connection reset by peer" error disappeared in a cmd shell when I first run `set "PATH=C:\Program Files\SSHFS-Win\bin"`.  Basically, even if I execute sshfs from within the SSHFS-Win\bin directory, without that path being set, it fails.  I suspect that SiriKali is running into the same issue here and I have to remember how to update the path in windows.

2.  When this account was first setup on this machine, it created the username using my full name including a space (... sigh).  Apparently sshfs is running into an issue when there's a space in the path that I haven't figured out how to escape so far.  

```"C:/Program Files/SSHFS-Win/bin\sshfs.exe" -f "cblakemore@----------:" "S:" -o rw,idmap=user,IdentityFile="C:/Users/Carl\ Blakemore/.ssh/id_rsa"
command-line line 0: garbage at end of line; "Blakemore/.ssh/id_rsa".
read: Connection reset by peer```

If I move the id_rsa file to a directory outside of my user directory and access it from that, I get successfully prompted for my passphrase and can connect.

All that being said, I suspect that if I fix the PATH, this problem should resolve itself since it will be using the socket.

**Comment by cblake777** [comment]
Total guess here, but I wonder if it has to do with `C:\WINDOWS\System32\OpenSSH\` being in the path, so when sshfs tries to run ssh, it's picking up the OpenSSH version, not the SSHFS-Win version.

**Comment by cblake777** [comment]
Confirmed.
```
C:\Users\Carl Blakemore>set PATH=C:\WINDOWS\System32\OpenSSH\;C:\Program Files\SSHFS-Win\bin

C:\Users\Carl Blakemore>"C:/Program Files/SSHFS-Win/bin\sshfs.exe" -f "cblakemore@--------:" "S:" -o rw,idmap=user,IdentityFile="C:/Unix/id_rsa"
read: Connection reset by peer

C:\Users\Carl Blakemore>set PATH=C:\Program Files\SSHFS-Win\bin;C:\WINDOWS\System32\OpenSSH\

C:\Users\Carl Blakemore>"C:/Program Files/SSHFS-Win/bin\sshfs.exe" -f "cblakemore@---------:" "S:" -o rw,idmap=user,IdentityFile="C:/Unix/id_rsa"
Enter passphrase for key 'C:/Unix/id_rsa':
The service sshfs has been started.
The service sshfs has been stopped.
```

**Comment by mhogomchungu** [comment]
SiriKali constructs PATH variable in a way that ensures the path to a backend is always ahead of system paths. The exact formular is: SIRIKALI_PATHS;BACKEND_PATH;WINDOWS_PATHS and below is what i am seeing here.
```
C:/Program Files (x86)/SiriKali\bin\;
C:/Users/Ink\bin\;
C:/Users/Ink\.bin\;
C:/Users/Ink/bin\;
C:\Program Files\SSHFS-Win\\bin\;
C:\Program Files\CryFS\0.10.2\bin;
C:\Windows\system32;C:\Windows;
C:\Windows\System32\Wbem;
C:\Windows\System32\WindowsPowerShell\v1.0\;
C:\Windows\System32\OpenSSH\;
C:\Program Files (x86)\Git\cmd;
C:\Program Files (x86)\CMake\bin;
C:\Program Files\PuTTY\;
C:\Users\Ink\AppData\Local\Microsoft\WindowsApps;
```

If the problem was a wrong version of ssh was getting picked, windows' ssh would exit with ```read: Connection reset by peer``` error like in your test and SiriKali would report this error and not report a "crashed" error you are seeing.

I just remembered and your problem is exactly like this one[1]. In both cases, sshfs is either:
1. Exiting for some reason without producing any logs.
2. SiriKali can't find it. This is unlikely because SiriKali would complain about it.
3. SiriKali can't run it.

[1] https://github.com/evsar3/sshfs-win-manager/issues/18

**Comment by cblake777** [comment]
Sorry, accidentally commented on the wrong issue.  I did run into one command line that caused sshfs.exe to return without doing anything.

The command was:

"C:/Program Files/SSHFS-Win/bin\sshfs.exe" -f "cblakemore@-------:" "S:"  -o rw,fsname="dh04",subtype=sshfs,idmap=user,StrictHostKeyChecking=no,IdentityFile="C:/Users/Carl Blakemore/.ssh/id_rsa",volname="cblakemore@node1.foobar.example.com"
I didn't think about it at the time because I was focused on making the ssh connection happen. I went back and played with it a bit more, the issue is volname. When I reduce the number of characters in volname to cblakemore@node1.foobar.example., then sshfs works and complains about the space in the IdentityFile path.

**Comment by mhogomchungu** [comment]
Yes, i can finally reproduce the issue, sshfs returns without saying anything if "volname" option has more than 32 characters and i have reported the issue here[1].

Can you add "volname=xxx" where "xxx" is less than 32 characters and check if you can finally connect using keepass?


[1] https://github.com/billziss-gh/winfsp/issues/311

**Comment by cblake777** [comment]
Indeed! SiriKali can now successfully mount the drive using KeePass!

Thanks!

**Comment by mhogomchungu** [comment]
I have released a new version(1.4.4-1) that contain the fix and i recommend you switch to it

https://github.com/mhogomchungu/sirikali/releases
